ts/features/wallet/component/FeaturedCardCarousel.tsx

```diff
@@ -0,0 +1,113 @@
+import { navigationHistoryPop } from "../../../store/actions/navigationHistory";
+import { availableBonusTypesSelector } from "../../bonus/bonusVacanze/store/reducers/availableBonusesTypes";
+import FeaturedCard from "./FeaturedCard";
+
+type Props = {
+  bvActive: boolean;
+} & ReturnType<typeof mapStateToProps> &
+  ReturnType<typeof mapDispatchToProps>;
+
+const FeaturedCardCarousel: React.FunctionComponent<Props> = (props: Props) => {
+  const handlersMap: Map<number, (bonus: BonusAvailable) => void> = new Map<
+    number,
+    (bonus: BonusAvailable) => void
+  >([[ID_BONUS_VACANZE_TYPE, bonus => props.navigateToBonusRequest(bonus)]]);
+
+  if (bpdEnabled) {
+    handlersMap.set(ID_BPD_TYPE, _ => props.startBpdOnboarding());
+  }
+
+  const isOneBonusActive = !props.bvActive || !getValue(props.bpdActiveBonus);
+  return isOneBonusActive ? (
+    <>
+      <View
+        style={[
+          IOStyles.horizontalContentPadding,
+          { paddingTop: 14, paddingBottom: 4 }
+        ]}
+      >
+        <H3 weight={"SemiBold"} color={"bluegreyDark"}>
+          {"In evidenza"}
+        </H3>
+      </View>
+      <ScrollView
+        horizontal={true}
+        style={[IOStyles.horizontalContentPadding, { paddingVertical: 10 }]}
+        alwaysBounceHorizontal={false}
+        showsHorizontalScrollIndicator={false}
+      >
+        {reverse([...props.availableBonusesList]).map((b, i) => {
+          const handler = fromNullable(handlersMap.get(b.id_type)).getOrElse(
+            () => nullType
+          );
+          switch (b.id_type) {
+            case ID_BONUS_VACANZE_TYPE:
+              return props.bvActive ? null : (
+                <FeaturedCard
+                  key={`featured_bonus_${i}`}
+                  title={b[getLocalePrimaryWithFallback()].name}
+                  image={b.cover}
+                  isNew={false}
+                  onPress={() => handler(b)}
+                />
+              );
+            case ID_BPD_TYPE:
+              return getValue(props.bpdActiveBonus) ? null : (
+                <FeaturedCard
+                  key={`featured_bonus_${i}`}
+                  title={I18n.t("bonus.bpd.name")}
+                  image={b.cover}
+                  isNew={true}
+                  onPress={() => handler(b)}
+                />
+              );
+            default:
+              return null;
+          }
+        })}
+      </ScrollView>
+    </>
+  ) : null;
+};
+
+const mapStateToProps = (state: GlobalState) => ({
+  bpdActiveBonus: bpdEnabledSelector(state),
+  availableBonusesList: pot.getOrElse(availableBonusTypesSelector(state), [])
+});
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  startBpdOnboarding: () => dispatch(bpdOnboardingStart()),
+  navigateToBonusRequest: (bonusItem: BonusAvailable) => {
+    dispatch(navigateToBonusRequestInformation({ bonusItem }));
+    dispatch(navigationHistoryPop(1));
```

Comment:
Is it needed? if I choose "bonus vacanze", when I press back the next screen is `MessageHomeScreen` instead of `WalletHomeScreen`. Removing that navigation pop it seems to work properly

Reply:
You're right! We have the same issue starting the bpdOnboarding and going back from it's detail page, i'm fixing in next commit!